File: library/src/main/java/com/getstream/sdk/chat/adapter/MessageListItemDiffCallback.java
package com.getstream.sdk.chat.adapter;


import androidx.recyclerview.widget.DiffUtil;

import java.util.List;

public class MessageListItemDiffCallback extends DiffUtil.Callback {

    private static final String TAG = MessageListItemDiffCallback.class.getSimpleName();

    protected List<MessageListItem> oldList, newList;

    public MessageListItemDiffCallback(List<MessageListItem> oldList, List<MessageListItem> newList) {
        this.oldList = oldList;
        this.newList = newList;
    }

    @Override
    public int getOldListSize() {
        return oldList == null ? 0 : oldList.size();
    }

    @Override
    public int getNewListSize() {
        return newList == null ? 0 : newList.size();
    }

    @Override
    public boolean areItemsTheSame(int oldItemPosition, int newItemPosition) {
        return oldList.get(oldItemPosition).equals(newList.get(newItemPosition));
    }

    @Override
    public boolean areContentsTheSame(int oldItemPosition, int newItemPosition) {
        return oldList.get(oldItemPosition).equals(newList.get(newItemPosition));
    }
}

File: library/src/main/java/com/getstream/sdk/chat/adapter/MessageListItemDiffCallback.kt
package com.getstream.sdk.chat.adapter

import androidx.recyclerview.widget.DiffUtil
import io.getstream.chat.android.livedata.utils.MessageListItem

object MessageListItemDiffCallback : DiffUtil.ItemCallback<MessageListItem>() {
    override fun areItemsTheSame(oldItem: MessageListItem, newItem: MessageListItem): Boolean {
        return oldItem.getStableId() == newItem.getStableId()
    }

    override fun areContentsTheSame(oldItem: MessageListItem, newItem: MessageListItem): Boolean {
        return oldItem == newItem
    }
}
